### PR TITLE
LPS-151251 Create URLBuilder to prep converting HttpComponentsUtil param methods into builder pattern

### PR DESCRIPTION
**Pull request by julschong**
@tinatian 
Hi Tina,

Main Task: [LPS-150839 Remove petra-apache-http-components](https://issues.liferay.com/browse/LPS-150839)
Sub Task: [LPS-151251 Create URLBuilder interface to provide builder interface to HttpComponentUtil](https://issues.liferay.com/browse/LPS-151251)

URLBuilder has no logic currently. Only using HttpComponentsUtil's logic.
Next part for this task is to move the logic to URLBuilder and apply to usages.

Thank you!

### CODE REVIEW

**Comment by julschong** [comment]
@catarineaguiar
Hi Catarine,
I got the LPS ticket number mixed up yesterday
This pull is the same as #921, just changed the LPS number

Thank you!

File: portal-kernel/src/com/liferay/portal/kernel/url/URLBuilder.java [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+		return setParameter(name, String.valueOf(value));
+	}
+
+	public URLBuilder setParameter(String name, short value) {
+		return setParameter(name, String.valueOf(value));
+	}
+
+	public URLBuilder setParameter(String name, String value) {
+		_url = HttpComponentsUtil.setParameter(_url, name, value);
+
```
Comment (julschong) [inline portal-kernel/src/com/liferay/portal/kernel/url/URLBuilder.java:96]: changed here to not include any logic from HttpComponentsUtil yet

**Comment by julschong** [comment]
ci:test:sf

**Comment by julschong** [comment]
ci:test:relevant

**Comment by julschong** [comment]
ci:test

**Comment by catarineaguiar** [comment]
@julschong semver error:

```
 [exec] Semantic versioning is incorrect while checking /opt/dev/projects/github/liferay-portal/portal-kernel/portal-kernel.jar against /opt/dev/projects/github/liferay-portal/.gradle/caches/modules-2/files-2.1/com.liferay.portal/com.liferay.portal.kernel/57.1.0/72710a4c6eb9b2b5c992bc6a86a2fc3a2829f9f2/com.liferay.portal.kernel-57.1.0.jar
```
full log [here](https://testray.liferay.com/reports/production/logs/2022-06/test-1-26/test-portal-acceptance-pullrequest(master)/3420/semantic-versioning-jdk8/0/0/jenkins-console.txt.gz?authuser=0)

**Comment by julschong** [comment]
pushed to update semver

**Comment by julschong** [comment]
ci:test:sf

**Comment by julschong** [comment]
ci:test:acceptance

**Comment by catarineaguiar** [comment]
@julschong no regressions.

**Pull request closed without merging**